src/main/java/bio/terra/workspace/db/DataReferenceDao.java

```diff
@@ -32,70 +30,63 @@
 
   private final NamedParameterJdbcTemplate jdbcTemplate;
-  private ObjectMapper objectMapper;
+  /**
+   * Database JSON ObjectMapper. Should not be shared with request/response serialization. We do not
+   * want necessary changes to request/response serialization to change what's stored in the
+   * database and possibly break backwards compatibility.
+   */
+  private static final ObjectMapper objectMapper = new ObjectMapper();
 
   @Autowired
-  public DataReferenceDao(NamedParameterJdbcTemplate jdbcTemplate, ObjectMapper objectMapper) {
+  public DataReferenceDao(NamedParameterJdbcTemplate jdbcTemplate) {
     this.jdbcTemplate = jdbcTemplate;
-    this.objectMapper = objectMapper;
   }
 
   private Logger logger = LoggerFactory.getLogger(DataReferenceDao.class);
 
-  public String createDataReference(
-      UUID referenceId,
-      UUID workspaceId,
-      String name,
-      UUID resourceId,
-      String credentialId,
-      CloningInstructionsEnum cloningInstructions,
-      ReferenceTypeEnum referenceType,
-      DataRepoSnapshot reference) {
+  /** Create a data reference in a workspace and return the reference's ID. */
+  public String createDataReference(DataReferenceRequest request, UUID referenceId)
+      throws DuplicateDataReferenceException {
     String sql =
-        "INSERT INTO workspace_data_reference (workspace_id, reference_id, name, resource_id, credential_id, cloning_instructions, reference_type, reference) VALUES "
-            + "(:workspace_id, :reference_id, :name, :resource_id, :credential_id, :cloning_instructions, :reference_type, cast(:reference AS json))";
+        "INSERT INTO workspace_data_reference (workspace_id, reference_id, name, cloning_instructions, reference_type, reference) VALUES "
+            + "(:workspace_id, :reference_id, :name, :cloning_instructions, :reference_type, cast(:reference AS json))";
 
     MapSqlParameterSource params =
         new MapSqlParameterSource()
-            .addValue("workspace_id", workspaceId.toString())
+            .addValue("workspace_id", request.workspaceId().toString())
             .addValue("reference_id", referenceId.toString())
-            .addValue("name", name)
-            .addValue("cloning_instructions", cloningInstructions.toString())
-            .addValue("credential_id", credentialId)
-            .addValue("resource_id", resourceId == null ? null : resourceId.toString())
-            .addValue("reference_type", referenceType == null ? null : referenceType.toString());
-    try {
-      params.addValue("reference", objectMapper.writeValueAsString(reference));
-    } catch (JsonProcessingException e) {
-      // TODO: add logger and print out the reference
-      throw new InvalidDataReferenceException(
-          "Couldn't convert reference to JSON. This... shouldn't happen.");
-    }
+            .addValue("name", request.name())
+            .addValue("cloning_instructions", request.cloningInstructions().toString())
```

Comment:
Might as well, that's probably the best practice. I don't use that for Stairway serialization as that just relies on ObjectMapper, and because we don't persist those for very long.